src/exporters/file/parquet/metric.rs

```diff
@@ -0,0 +1,294 @@
+            .map_err(|e| FileExporterError::Export(e.to_string()))
+    }
+}
+
+impl MetricRow {
+    pub fn from_resource_metrics(
+        resource_metrics: &ResourceMetrics,
+    ) -> Result<Vec<MetricRow>, FileExporterError> {
+        // retain core imports only
+        use opentelemetry_proto::tonic::common::v1::any_value::Value as AnyValue;
+
+        fn attrs_to_map(attrs: &[opentelemetry_proto::tonic::common::v1::KeyValue]) -> MapOrJson {
+            let mut map = std::collections::HashMap::new();
+            for attr in attrs {
+                if let Some(any_value) = &attr.value {
+                    let value_str = match &any_value.value {
+                        Some(AnyValue::StringValue(s)) => s.clone(),
+                        Some(AnyValue::BoolValue(b)) => b.to_string(),
+                        Some(AnyValue::IntValue(i)) => i.to_string(),
+                        Some(AnyValue::DoubleValue(d)) => d.to_string(),
+                        _ => "".to_string(),
+                    };
+                    map.insert(attr.key.clone(), value_str);
+                }
+            }
+            MapOrJson::Map(map)
+        }
+
+        // Resource-level attributes ----------------------------
+        let resource_attrs = resource_metrics
+            .resource
+            .as_ref()
+            .map(|r| attrs_to_map(&r.attributes))
+            .unwrap_or(MapOrJson::Map(Default::default()));
+
+        // Extract service.name from resource attributes
+        let service_name = if let MapOrJson::Map(map) = &resource_attrs {
+            map.get("service.name").cloned().unwrap_or_default()
+        } else {
+            String::new()
+        };
+
+        let mut rows = Vec::new();
+
+        for scope_metrics in &resource_metrics.scope_metrics {
+            let scope_name = scope_metrics
+                .scope
+                .as_ref()
+                .map(|s| s.name.clone())
+                .unwrap_or_default();
+            let scope_version = scope_metrics
+                .scope
+                .as_ref()
+                .map(|s| s.version.clone())
+                .unwrap_or_default();
+            let scope_attributes = scope_metrics
+                .scope
+                .as_ref()
+                .map(|s| attrs_to_map(&s.attributes))
+                .unwrap_or(MapOrJson::Map(Default::default()));
+
+            for metric in &scope_metrics.metrics {
+                let type_ = match metric.data.as_ref() {
+                    Some(Data::Gauge(_)) => "gauge",
+                    Some(Data::Sum(_)) => "sum",
+                    Some(Data::Histogram(_)) => "histogram",
+                    Some(Data::ExponentialHistogram(_)) => "exponential_histogram",
+                    Some(Data::Summary(_)) => "summary",
+                    None => "unknown",
+                };
+
+                // We treat the *first* data point as representative for timestamp & value.
```

Comment:
This seems like we would lose data here. Should we generate a row per data point?

Reply:
updated to export all data points